### ISSUE TITLE
Module clash exception needs checkBuildPlan to be aware allowed clashes in yaml file.

### ISSUE THREAD

**Issue by drwebb** [fpco/stackage-types#1]
@snoyberg 

I've made two commits to stackage curator

1) To add Tests and possibility for module clash exception 
https://github.com/drwebb/stackage-curator/commit/00844d59203d78b308a28d2ff51802fbd550d3be

2) To fix all the name issues I introduced:
https://github.com/drwebb/stackage-curator/commit/795ccfda0c4b9de460c3f37a7ac0c7fd9e4f58db

`checkBuildPlan` and `updateBuildPlan` in stackage-curator need to source the values that are present in the build constraints yaml file, used to generate `PackageContraints` and `BuildContraints`.

Does it look okay to add as an `Map ExportedModuleName (Set PackageName)` field the BuildPlan type as a field? You can see as reference the `BuildConstraints` type:

``` haskell
data BuildConstraints = BuildConstraints
    { bcPackages           :: Set PackageName
    -- ^ This does not include core packages.
    , bcPackageConstraints :: PackageName -> PackageConstraints

    , bcSystemInfo         :: SystemInfo

    , bcGithubUsers        :: Map Text (Set Text)
    -- ^ map an account to set of pingees
    , bcAllowedModuleClashes :: Map Text (Set Text)
    -- ^ map from module names to packages that contain the same exported name
    }
```

**Comment by snoyberg** [comment]
Yes, with the following caveat: I think it should be a `Maybe` field. The semantics should be: if the value is not present in the YAML file, then checking is disabled. This way, when working on a preexisting snapshot (like lts-2.8), we don't suddenly get failures mid-major release. Does that make sense?

**Comment by drwebb** [comment]
Right makes perfect sense, it's such a small one line change I'll leave making the pull request for this repo when I push the rest of the code to stackage-curator.
